handler/core/src/main/java/io/knotx/fragments/handler/FragmentsHandler.java

```diff
@@ -80,13 +79,13 @@ public void handle(RoutingContext routingContext) {
 
     boolean isDebugMode = isDebugModeOn(clientRequest);
     List<FragmentEventContextTaskAware> events = toEvents(fragments, clientRequest);
-    if (isDebugMode) {
-      events.forEach(debugModeDecorator::markAsDebuggable);
-    }
+    FragmentsDebugModeDecorator debugModeDecorator = debugModeDecoratorFactory.create(events, isDebugMode);
```

Comment:
I'd like that optimally no (or very little) objects are created when `debugMode=false`.
There is no need to trash the memory with multiple objects what do nothing.

Reply:
check how do you like new approach